### PR TITLE
make yum.tfm.o cache rpm (and compressed repodata) for a month

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: puppet/modules/web/manifests/init.pp [hunk 1]
```diff
@@ -84,10 +84,24 @@
...
 
   # YUM
+  $yum_directory = '/var/www/vhosts/yum/htdocs'
+  $yum_directory_config = [
+    {
+      path    => $yum_directory,
+      options => ['Indexes', 'FollowSymLinks', 'MultiViews'],
+    },
+    {
+      path     => '.+\.(bz2|gz|rpm)$',
```
Comment (ekohl) [inline puppet/modules/web/manifests/init.pp]: Do we have any `.xz` files?

File: puppet/modules/web/manifests/init.pp [hunk 2]
```diff
@@ -84,10 +84,24 @@
...
+    },
+  ]
+
   $yum_attrs = {
     servername      => 'yum.theforeman.org',
-    docroot         => '/var/www/vhosts/yum/htdocs',
+    docroot         => $yum_directory,
     docroot_mode    => '2575',
+    directories     => $yum_directory_config,
     custom_fragment => template('web/yum.conf.erb'),
```
Comment (ekohl) [inline puppet/modules/web/manifests/init.pp]: In this file we use mod_expires. Can we unify this and add `repomd.xml` to `$yum_directory_config`?

File: puppet/modules/web/manifests/init.pp (hunk 1 above)
Comment (evgeni) [reply puppet/modules/web/manifests/init.pp]: I didn't `find(1)` any, but can still be added for future-proofing if you want?

File: puppet/modules/web/manifests/init.pp (hunk 2 above)
Comment (evgeni) [reply puppet/modules/web/manifests/init.pp]: We'd loose the `IfModule`, but otherwise, yeah

Comment (evgeni) [reply puppet/modules/web/manifests/init.pp]: After a second look, no, it's not that easy.

Comment (evgeni) [reply puppet/modules/web/manifests/init.pp]: (don't trust the docs)

**Pull request merged**